docs/api-types.md

```diff
@@ -0,0 +1,100 @@
+    -d '
+    {
+      "contentsHash":"01a7cc7de66f6e398ccc5bfdeb2db81646aed6e43f90f7333d9a06ba205b7f74",
+      "assignmentHash":"219bfb1e71af4f888fce709900862b4467f45b42b0603a56e9428cb3e8eb8940",
+      "witness":"83005820836d6a5908166c398db194e7af916827f63da021a6a3a1c3edf9ced6608adf2b5840d73bbe45a710b4430e085aa6c1cf374417f5fc1a97121945ef2611a82346cea675bdc63d14f1c6f69b64438cd31e2e0bb17d61ecad4ef17c4d04f297857b6803"
+    }
+'
+```
+(Try it on launched [educator node with bot](/educator/README.md#scripts)).
+
+In detail, request for this endpoint is formed as follows:
+
+```javascript
+function makeSignedSubmission(
+    studentSecretKey, // student's secret key
+    contentsHash      = "7b6d0c6de38639cc6063e9c36f9dcdb71fff60fe551ccc757246a3bf2fa00f37",
+                      // hash of submitted solution
+    assignmentHash    // assignment identifier, returned along with other information about an assignment
+) {
+    return
+        { contentsHash: contentsHash
+        , assignmentHash: assignmentHash
+        , witness: makeWitness(studentSecretKey, contentsHash, assignmentHash)
+        }
+}
+```
+
+If submission does not contain solution, i.e. submission is [offline](/specs/disciplina/educator/api/spec.org) one, then for `contentsHash` parameter default value should be used.
+
+`Witness` field validates the fact that submission was made by its claimed author and is constructed as follows.
+
+```javascript
+function makeWitness(
+    studentSecretKey, // student's secret key
+    contentsHash,     // hash of submitted solution
+                      // (if submission is offline one, use dedicated
+                      // "7b6d0c6de38639cc6063e9c36f9dcdb71fff60fe551ccc757246a3bf2fa00f37" value)
+    assignmentHash    // assignment identifier, returned along with other information about an assignment
+) {
+    // Used libraries
+    var cbor;     // encodes entities according to CBOR specification, see also ??? (TODO: DSCP-267)
+    var ed22519;  // operations on Ed22519 keys
+
+    // Algorithm
+    var studentPublicKey = ed22519.toPublic(studentSecretKey)
+    var studentAddress = blake2b(studentPublicKey, { digestLength: 16 })
+    var submission = cbor.serialise([studentAddress, contentsHash, assignmentHash])
+    var submissionHash = blake2b(submission, { digestLength: 16 })
+    var signature = ed22519.sign(studentSecretKey, submissionHash)
+    var witness = cbor.serialise([studentPublicKey, signature])
+    return toHex(witness)
+}
+```
+
+#### Evaluation example
+
+Let's assume we have following data (all entities are given in hex for convenience, algorithm itself operates with raw bytes):
+
+```javascript
+var studentSecretKey = "6dbb1eeeb3e8e81547f307ba9344dae6286a9319458e9596f2ffc43670d5c648"
+var contentsHash = "01a7cc7de66f6e398ccc5bfdeb2db81646aed6e43f90f7333d9a06ba205b7f74"
+var assignmentHash = "219bfb1e71af4f888fce709900862b4467f45b42b0603a56e9428cb3e8eb8940"
+```
+
+Then algorithm proceedes as follows:
```

Comment:
`proceeds`

Reply:
I have a brilliant idea: let's introduce a requirement that documentation should be passed through grammarly.com before submitting :thinking:

Reply:
That sounds like a really good idea. Although I'm not sure how can we submit a document on check quickly.